framework/0008-naming-conventions.md

```diff
@@ -0,0 +1,71 @@
+- Start Date: 2019-10-03
+- RFC PR: [8](https://github.com/inveniosoftware/rfcs/pull/8)
+- Authors:
+
+# Naming Conventions
+
+> There are only two hard things in Computer Science: cache invalidation and naming things.
+> - Phil Karlton
+
+
+## Summary
+
+Here are *recommendations* for naming things in Invenio.
+
+## Motivation
+
+As a developer, I want names to be consistent and have help in naming
+things, so that I can name, parse and use modules and configuration variables
+without namespace clashes (especially on public package indices), by relying on
+these conventions to make automation tools and to not have to go through
+bike-shedding whenever I name things.
+
+We think [convention over configuration](https://rubyonrails.org/doctrine/#convention-over-configuration)
+in naming is a good approach to cut down on mundane recurring decisions. This
+RFC will suffer so you don't have to.
+
+## Detailed design
+
+Any installable Invenio module, should be named `invenio-<module>`. This naming
+pattern can be snake_cased directly when used in Python code: `invenio_<module>`.
+
+Module names should be plural when about countable thing e.g.,
+`invenio-records`, `invenio-stats`, `invenio-communities`.
+
+The `<module>` part of the module name should be used to prefix the configuration
+variables of the modules. Those are always in CAPITAL_SNAKE_CASE. For example,
+for the `invenio-bar-foos` module: `BAR_FOOS_<CONFIGURATION_VARIABLE>` is used
+in `config.py`
+
+Increasingly specific sub-projects should prefix the `<module>` part:
+`invenio-rdm-<module>`, `invenio-records-<module>` (e.g., `invenio-records-rest`,
+`invenio-records-permissions`)
+
+Non-Invenio projects part of another framework should follow that framework's
+conventions. (e.g., `flask-menu` for a flask module).
+
+Other than that (general libraries beyond Invenio, e.g. `datacite`), follow the
+conventions of the language you are using.
+
+
+## How we teach this
+
+In contributor documentation, we point to this RFC.
+
+
+## Drawbacks
+
+To respect these conventions, a couple of modules would need to be renamed:
+
+- `invenio-deposit` -> `invenio-deposits`
```

Comment:
This would be hard. Maybe it can be done in the future if a refactor is needed. However, we can apply the conventions from now on.

Reply:
I wonder if there is a way on PyPi to have a "package redirect"...

Reply:
Fair, "eventually" is good :angel:. We can open an issue to investigate the package redirect at some point. However, the main issue is the API, whoever programming on invenio sill have `invenio_desposit` everywhere in the imports. But yeah it can be done eventually :+1: